### PR TITLE
UI: Balance component

### PR DESCRIPTION
**Pull request by kpyszkowski**
Ref: #2
Mentioned in: #15 
~Depends on:#12~
This pull request introduces `TokenBalanceCardTemplate` component redesign.
localhost_3000_tBTC_mint (5)

### CODE REVIEW

File: .env.example
Comment (michalsmiarowski) [inline .env.example:1]: Let's revert changes in this file

File: src/components/TokenBalanceCard/TokenBalanceCardTemplate.tsx
Comment (michalsmiarowski) [inline src/components/TokenBalanceCard/TokenBalanceCardTemplate.tsx:12]: Some of those import are not used. Let's remove them

File: src/components/TokenBalanceCard/TokenBalanceCardTemplate.tsx
Comment (michalsmiarowski) [inline src/components/TokenBalanceCard/TokenBalanceCardTemplate.tsx:64]: This is the only place where we use `TokenBalance` component (but please double-check this), so we could remove all the properties that are not used anymore (like `isLarge for example).

We could also move the `tokenSymbol` text below inside `TokenBalance` and replace the old one here, but I'll leave that decision to you. I think it would make sense to keep it closely tied to the `TokenBalance` component next to it's amount. If you're not gonna replace it then we can also remove `withSymbol` and `tokenSymbol` properties from `TokenBalance`. Please check for other unused properties and remove them 😉

**Review by michalsmiarowski — CHANGES_REQUESTED** [review]
Left some comments down below. Let's also merge `main` branch here and resolve the conflicts.

File: src/components/TokenBalanceCard/TokenBalanceCardTemplate.tsx
Comment (kpyszkowski) [reply src/components/TokenBalanceCard/TokenBalanceCardTemplate.tsx]: image

There is actually large variant of the component I forgot to account. I introduced it with this commit: 3623cd1b

Also I've removed redundant props like boolean flags that can be defined directly by the presence of the value.
Example: to display values with higher precision you have to define both `higherPrecision` prop which is pretty much self explanatory and `withHigherPrecision` prop - which is counter intuitive. If I define higher precision, I'd like to display such value, that said the boolean `withHigherPrecision` is redundant.

File: src/pages/tBTC/Bridge/TbtcBalanceCard.tsx
Comment (michalsmiarowski) [inline src/pages/tBTC/Bridge/TbtcBalanceCard.tsx:15]: Now I see something problematic about removing the `withHigherPrecision` flag. The way it worked before was that the `higherPrecision` is an optional value, and if it's not provided, the default value of it would be 2 (it is set in `InlineTokenBalance`). Now, you could pass `withHigherPrecision` flag to component and it would still display the tooltip (that is placed in `InlineTokenBalance`) even if you don't pass the `precision` prop. I agree that it is counter-intuitive though.

Because of the thing that I'v described above, the higher precision tooltip is not displayed for tbtc balance that is in the top layout section.

(1st solution) We can fix that by passing `higherPrecision={2}` to `TokenBalanceCard` in `TbtcBalanceCard`. By removing the `withHigherPrecision` property we will have to always pass a `higherPrecision` property to display the tooltip, which too might be little counter-intuitive, as we would have to think about what number should be passed there instead just passing a true value and let the component handle it itself, but I think this solution would be good enough.

```suggestion
      higherPrecision={2}
```

(2nd solution) We could also create additional property, which would be `withHigherPrecisionTooltip` and if it's set to true we could display the tooltip in `InlineTokenBalance`. So, in `TbtcBalanceCard` component:

```
{shouldRenderTokenAmount ? (
            (withHigherPrecisionTooltip || precision) ? (
              <InlineTokenBalance
                tokenAmount={tokenAmount}
                tokenFormat={tokenFormat}
                tokenDecimals={tokenDecimals}
                precision={precision}
                withHigherPrecision
              />
            ) : (
              _tokenAmount
            )
          ) : (
            "--"
          )}
```

I think this is why we've had `withHigherPrecision` property in the first place. We would not need to pass the `higherPrecision` property then.

I'll let you decide which fix you prefer. Both are ok.

image

**Review by michalsmiarowski — CHANGES_REQUESTED** [review]
Left some changes to look at before the merge

File: src/components/TokenBalanceCard/index.tsx
Comment (kpyszkowski) [reply src/components/TokenBalanceCard/index.tsx:40]: Actually the `withUsdBalance` here is needed. There are use cases of this component where we use this component with or without the USD balance so we need to keep it toggleable.

With balance:
image
Without balance:
image

File: src/components/TokenBalanceCard/index.tsx
Comment (kpyszkowski) [reply src/components/TokenBalanceCard/index.tsx:30]: Let's do it in a future

File: src/pages/tBTC/Bridge/TbtcBalanceCard.tsx
Comment (kpyszkowski) [reply src/pages/tBTC/Bridge/TbtcBalanceCard.tsx]: Ref commit: 87358e2d

File: src/components/TokenBalance.tsx
Comment (kpyszkowski) [reply src/components/TokenBalance.tsx:82]: Ref commit: 35c273de

File: src/components/TokenBalanceCard/TokenBalanceCardTemplate.tsx
Comment (kpyszkowski) [reply src/components/TokenBalanceCard/TokenBalanceCardTemplate.tsx]: Ref commit: 35c273de

File: src/components/TokenBalanceCard/index.tsx
Comment (michalsmiarowski) [reply src/components/TokenBalanceCard/index.tsx:37]: Ah, yes, you are right! We get the `usdBalance` prop from the state and we only use it when `withUsdBalance` is set to true.

My bad 🙊 But good catch tho 🔥

File: src/components/TokenBalance.tsx
Comment (michalsmiarowski) [inline src/components/TokenBalance.tsx:112]: We should pass `precision` here and add `higherPrecision` too

```suggestion
                precision={precision}
                higherPrecision={higherPrecision}
```

File: src/pages/tBTC/Bridge/TbtcBalanceCard.tsx
Comment (michalsmiarowski) [inline src/pages/tBTC/Bridge/TbtcBalanceCard.tsx:14]: I've made a mistake and we actually should use 6 here based on this:

https://github.com/threshold-network/bitcoin-on-base/blob/70e7582866a3aff92066b85eec4f2872811eaedb/src/components/TokenBalance.tsx#L40

```suggestion
      higherPrecision={6}
```

File: src/components/TokenBalance.tsx
Comment (michalsmiarowski) [inline src/components/TokenBalance.tsx:95]: Let's add `precision` to `useMemo` dependencies:

```suggestion
  }, [tokenAmount, tokenFormat, tokenDecimals, dependencies])
```

File: src/components/TokenBalanceCard/TokenBalanceCardTemplate.tsx
Comment (michalsmiarowski) [inline src/components/TokenBalanceCard/TokenBalanceCardTemplate.tsx:57]: I think `isEstimated` is not needed here. The `TokenBalance` component doesn't have `isEstimated` prop so this results in this warning in console:

```
Warning: React does not recognize the `isEstimated` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `isestimated` instead. If you accidentally passed it from a parent component, remove it from the DOM element.
```

Either way, `isEstimated` property should not be set to true in case to tbtc value as it is real value, not estimated. The only case where it will use the `~` mark should be when the precision (or higher precision in case of tooltip) is smaller than the real value. I think it does not work as expected right now (it only works if the value is smaller than 1) but this is something that we should fix in a future PR.

For now, let's just remove that `isEstimated` prop that is being passed to `TokenBalance`

**Review by michalsmiarowski — CHANGES_REQUESTED** [review]
Left few more comments

File: src/pages/tBTC/Bridge/TbtcBalanceCard.tsx
Comment (michalsmiarowski) [reply src/pages/tBTC/Bridge/TbtcBalanceCard.tsx]: Conversation continued here: https://github.com/threshold-network/bitcoin-on-base/pull/17#discussion_r1489804397

File: src/components/TokenBalanceCard/TokenBalanceCardTemplate.tsx
Comment (kpyszkowski) [reply src/components/TokenBalanceCard/TokenBalanceCardTemplate.tsx]: Ref commit: 90c7608b

File: src/components/TokenBalance.tsx
Comment (kpyszkowski) [reply src/components/TokenBalance.tsx]: Ref commit: 795d533a

File: src/pages/tBTC/Bridge/TbtcBalanceCard.tsx
Comment (kpyszkowski) [reply src/pages/tBTC/Bridge/TbtcBalanceCard.tsx]: Ref commit: ee97ceac

File: src/components/TokenBalance.tsx
Comment (kpyszkowski) [reply src/components/TokenBalance.tsx:112]: Ref commit: b11bf6cb

File: src/components/TokenBalance.tsx
Comment (kpyszkowski) [reply src/components/TokenBalance.tsx]: Ref commit: a60ec7a7

File: src/components/TokenBalance.tsx
Comment (michalsmiarowski) [reply src/components/TokenBalance.tsx:112]: I think you forgot to change `precision={higherPrecision}` to `precision={precision}`. 

Should be:
```
  precision={precision}
  higherPrecision={higherPrecision}
```

Right now it's:
```
  precision={higherPrecision}
  higherPrecision={higherPrecision}
```

Comment (kpyszkowski) [reply src/components/TokenBalance.tsx]: Ref commit: 33dbe121

**Review by michalsmiarowski — APPROVED** [review]
LGTM

**Pull request merged**
